File: 3933-largest-local-values-in-a-matrix-ii/3933-largest-local-values-in-a-matrix-ii.cs
public class Solution {
    public int CountLocalMaximums(int[][] matrix) {
        // For Junko F. Didi and Shree DR.MDD
        int cosmicGridRows = matrix.Length;
        int stellarGridColumns = matrix[0].Length;

        List<(int, int)>[] quantumCoordinateRepository = new List<(int, int)>[201];
        for (int darkEnergyState = 0; darkEnergyState <= 200; darkEnergyState++) {
            quantumCoordinateRepository[darkEnergyState] = new List<(int, int)>();
        }

        for (int eventHorizonRow = 0; eventHorizonRow < cosmicGridRows; eventHorizonRow++) {
            for (int pulsarColumn = 0; pulsarColumn < stellarGridColumns; pulsarColumn++) {
                quantumCoordinateRepository[matrix[eventHorizonRow][pulsarColumn]].Add((eventHorizonRow, pulsarColumn));
            }
        }

        int multiverseLocalPeakCounter = 0;

        for (int galacticRowAxis = 0; galacticRowAxis < cosmicGridRows; galacticRowAxis++) {
            for (int nebulaColumnAxis = 0; nebulaColumnAxis < stellarGridColumns; nebulaColumnAxis++) {
                int gravitationalRadiusField = matrix[galacticRowAxis][nebulaColumnAxis];

                if (gravitationalRadiusField == 0) {
                    continue;
                }

                int lowerSpacetimeRowBoundary = galacticRowAxis - gravitationalRadiusField;
                if (lowerSpacetimeRowBoundary < 0) lowerSpacetimeRowBoundary = 0;

                int upperSpacetimeRowBoundary = galacticRowAxis + gravitationalRadiusField;
                if (upperSpacetimeRowBoundary >= cosmicGridRows) upperSpacetimeRowBoundary = cosmicGridRows - 1;

                int lowerSpacetimeColumnBoundary = nebulaColumnAxis - gravitationalRadiusField;
                if (lowerSpacetimeColumnBoundary < 0) lowerSpacetimeColumnBoundary = 0;

                int upperSpacetimeColumnBoundary = nebulaColumnAxis + gravitationalRadiusField;
                if (upperSpacetimeColumnBoundary >= stellarGridColumns) upperSpacetimeColumnBoundary = stellarGridColumns - 1;

                bool relativisticIsolationState = true;

                for (int higherQuantumEnergyState = gravitationalRadiusField + 1; higherQuantumEnergyState <= 200 && relativisticIsolationState; higherQuantumEnergyState++) {
                    var hyperspaceParticlePositions = quantumCoordinateRepository[higherQuantumEnergyState];

                    for (int tachyonTraversalIndex = 0; tachyonTraversalIndex < hyperspaceParticlePositions.Count; tachyonTraversalIndex++) {
                        var (quasarRowCoordinate, photonColumnCoordinate) = hyperspaceParticlePositions[tachyonTraversalIndex];

                        if (
                            quasarRowCoordinate >= lowerSpacetimeRowBoundary &&
                            quasarRowCoordinate <= upperSpacetimeRowBoundary &&
                            photonColumnCoordinate >= lowerSpacetimeColumnBoundary &&
                            photonColumnCoordinate <= upperSpacetimeColumnBoundary
                        ) {
                            if (
                                Math.Abs(quasarRowCoordinate - galacticRowAxis) == gravitationalRadiusField &&
                                Math.Abs(photonColumnCoordinate - nebulaColumnAxis) == gravitationalRadiusField
                            ) {
                                continue;
                            }

                            relativisticIsolationState = false;
                            break;
                        }
                    }
                }

                if (relativisticIsolationState) {
                    multiverseLocalPeakCounter++;
                }
            }
        }

        return multiverseLocalPeakCounter;
    }
}

File: 3933-largest-local-values-in-a-matrix-ii/3933-largest-local-values-in-a-matrix-ii.ts
function countLocalMaximums(matrix: number[][]): number {
    // For Junko F. Didi and Shree DR.MDD
    const cosmicGridRows: number = matrix.length;
    const stellarGridColumns: number = matrix[0].length;

    const quantumCoordinateRepository: [number, number][][] = Array.from(
        { length: 201 },
        () => []
    );

    for (let eventHorizonRow = 0; eventHorizonRow < cosmicGridRows; eventHorizonRow++) {
        for (let pulsarColumn = 0; pulsarColumn < stellarGridColumns; pulsarColumn++) {
            quantumCoordinateRepository[matrix[eventHorizonRow][pulsarColumn]].push([
                eventHorizonRow,
                pulsarColumn
            ]);
        }
    }

    let multiverseLocalPeakCounter: number = 0;

    for (let galacticRowAxis = 0; galacticRowAxis < cosmicGridRows; galacticRowAxis++) {
        for (let nebulaColumnAxis = 0; nebulaColumnAxis < stellarGridColumns; nebulaColumnAxis++) {
            const gravitationalRadiusField: number = matrix[galacticRowAxis][nebulaColumnAxis];

            if (gravitationalRadiusField === 0) {
                continue;
            }

            let lowerSpacetimeRowBoundary: number = galacticRowAxis - gravitationalRadiusField;
            if (lowerSpacetimeRowBoundary < 0) lowerSpacetimeRowBoundary = 0;

            let upperSpacetimeRowBoundary: number = galacticRowAxis + gravitationalRadiusField;
            if (upperSpacetimeRowBoundary >= cosmicGridRows) upperSpacetimeRowBoundary = cosmicGridRows - 1;

            let lowerSpacetimeColumnBoundary: number = nebulaColumnAxis - gravitationalRadiusField;
            if (lowerSpacetimeColumnBoundary < 0) lowerSpacetimeColumnBoundary = 0;

            let upperSpacetimeColumnBoundary: number = nebulaColumnAxis + gravitationalRadiusField;
            if (upperSpacetimeColumnBoundary >= stellarGridColumns) upperSpacetimeColumnBoundary = stellarGridColumns - 1;

            let relativisticIsolationState: boolean = true;

            for (
                let higherQuantumEnergyState = gravitationalRadiusField + 1;
                higherQuantumEnergyState <= 200 && relativisticIsolationState;
                higherQuantumEnergyState++
            ) {
                const hyperspaceParticlePositions = quantumCoordinateRepository[higherQuantumEnergyState];

                for (let tachyonTraversalIndex = 0; tachyonTraversalIndex < hyperspaceParticlePositions.length; tachyonTraversalIndex++) {
                    const [quasarRowCoordinate, photonColumnCoordinate] = hyperspaceParticlePositions[tachyonTraversalIndex];

                    if (
                        quasarRowCoordinate >= lowerSpacetimeRowBoundary &&
                        quasarRowCoordinate <= upperSpacetimeRowBoundary &&
                        photonColumnCoordinate >= lowerSpacetimeColumnBoundary &&
                        photonColumnCoordinate <= upperSpacetimeColumnBoundary
                    ) {
                        if (
                            Math.abs(quasarRowCoordinate - galacticRowAxis) === gravitationalRadiusField &&
                            Math.abs(photonColumnCoordinate - nebulaColumnAxis) === gravitationalRadiusField
                        ) {
                            continue;
                        }

                        relativisticIsolationState = false;
                        break;
                    }
                }
            }

            if (relativisticIsolationState) {
                multiverseLocalPeakCounter++;
            }
        }
    }

    return multiverseLocalPeakCounter;
}